spec/unit/lib/cloud_controller/runner_spec.rb

```diff
@@ -66,6 +66,46 @@ module VCAP::CloudController
         end
       end
 
+      describe 'setting max db connections per process for puma' do
+        context 'when max db connections per process value is specified' do
+          before do
+            TestConfig.override(webserver: 'puma', puma: { max_db_connections_per_process: 10 })
+            allow(Config).to receive(:load_from_file).and_return(TestConfig.config_instance)
+          end
+
+          it 'overrides the db max_connections' do
+            db_config = subject.instance_variable_get(:@config).get(:db)
+            db_logger = Steno.logger('cc.db')
+
+            expect(subject.instance_variable_get(:@server)).to be_an_instance_of(PumaRunner)
+            expect(subject.config.get(:puma, :max_db_connections_per_process)).to eq(10)
+            expect(subject.config.get(:db, :max_connections)).to eq(10)
+            expect(DB).to receive(:load_models).with(db_config, db_logger)
+
+            subject.send(:setup_db)
```

Comment:
I'm in favor of precise test cases checking a single thing, thus I would leave out _expectations_ that are already checked elsewhere (e.g. `PumaRunner`) or not relevant (e.g. the `max_db_connections_per_process` config value). With the `hash_including` matcher you can explicitly check for the correctly set `max_connections` value.

```suggestion
            expect(DB).to receive(:load_models).with(hash_including(max_connections: 10), an_instance_of(Steno::Logger))

            subject
```